supabase/migrations/20230923153903_new_permit.sql

```diff
@@ -0,0 +1,48 @@
+CREATE TABLE IF NOT EXISTS
+  new_permits (
+    id serial primary key,
+    created timestamptz not null,
+    updated timestamptz not null,
+    network smallserial not null,
+    token varchar(42) not null,
```

Comment:
If it's only limited to the EVM token address, no need varchar, it can be only character(42).

Reply:
if we use character() then this [error](https://github.com/Seprintour-Test/test/issues/41#issuecomment-1732579005) happens

Reply:
We'd use text instead, with a max check.. should be more stable than the characters

Reply:
> if we use character() then this https://github.com/Seprintour-Test/test/issues/41#issuecomment-1732579005 happens

The root cause of this issue is not character because it's the data type being used a lot in postgresql type definition personally but there could be other factors. so once it gets merged, let me handle it. Lets skip it for now if you're make sure it gives error

Reply:
https://github.com/ubiquity/ubiquibot/blob/development/supabase/migrations/20230207011749_create_wallet_table.sql#L3
For the reference @whilefoo

Reply:
> The root cause of this issue is not character because it's the data type being used a lot in postgresql type definition in my previous use cases but there could be other factors. so once it gets merged, let me handle it. Lets skip it for now if you can make sure it gives an error

That's the error from using `char`, and it's known issue

Reply:
Can you take the example of the error case? Even though we used it correctly, it could throw an error. this is what you're saying right?

Reply:
> it could throw an error. this is what you're saying right?

Yea, right.. the schema has changed though and a work in progress, so I'm waiting on it

Reply:
>  a work in progress

do you know where we're tracking this? @whilefoo @seprintour

Reply:
It's here #787

Reply:
![image](https://github.com/ubiquity/ubiquibot/assets/108444211/ec8e98d8-b1e2-485b-8116-fa1e0a07b142)

Reply:
![image](https://github.com/ubiquity/ubiquibot/assets/108444211/58b5ae0c-0dfb-4c06-a4c3-4121628b349f)

Reply:
yeah. he seems right in type definition.

Reply:
If you're trying to store an exact 42-length string to the character(42), it should definitely work. but the length isn't equal to 42, it would fail to store string into the character(42). It is kind of an another sanitation check to use exact length.

Reply:
We changed `wallet_address` to text exactly because of this reason (https://github.com/ubiquity/ubiquibot/issues/620)
https://github.com/ubiquity/ubiquibot/blob/e3b4fea0ba3bc27f4e74ed60249cf41c67305592/supabase/migrations/20230826183500_change_wallet_type.sql#L1

Reply:
> We changed `wallet_address` to text exactly because of this reason (https://github.com/ubiquity/ubiquibot/issues/620)
> https://github.com/ubiquity/ubiquibot/blob/e3b4fea0ba3bc27f4e74ed60249cf41c67305592/supabase/migrations/20230826183500_change_wallet_type.sql#L1

Makes sense now @0xcodercrane , it was altered down in the migration